root/account/applications/Register.js

```diff
@@ -71,10 +71,11 @@ class RegisterApplication extends React.Component<Props, State> {
               label={l('Type:')}
               onChange={this.handleOauthTypeChange}
               options={oauthTypeOptions}
+              value={this.state.oauthType}
```

Comment:
Additional vain attempt to sync the selected value with an elaborate component state.

Reply:
My mental concept of the React form stuff is that the `form` should be stored in state (unless you're using redux, like in root/static/scripts/work.js), and you should update the form object so that the state (values, errors) stay in one place. One way to do that while keeping the state immutable is using the lens utility:

```JavaScript
const {Lens, prop, set, compose3} = require('../../static/scripts/common/utility/lens');

const oauthTypeFieldLens: Lens<ApplicationRegisterForm, OauthTypeT> =
  compose3(prop('field'), prop('oauth_type'), prop('value'));

// ...
  handleOauthTypeChange(e: SyntheticEvent<HTMLSelectElement>) {
    this.setState({
      form: set(oauthTypeFieldLens, (e.currentTarget.value: any), this.state.form),
    });
  }
// ...
```

The advantage is that we keep all of the form state in a single place, and the form components only have to look at the field they receive and not any overrides. The form/fields also store errors in addition to values, so it will make more sense once we're displaying dynamic field errors too.